// @tarojs/components/lib/react/component-lib/index.js
import { Fragment } from 'react';
import reactifyWebComponent from './reactify-wc.js';
export { default as Input } from './input.js';

// 视图容器
const CoverImage = reactifyWebComponent('taro-cover-image-core');
const CoverView = reactifyWebComponent('taro-cover-view-core');
const MatchMedia = reactifyWebComponent('taro-match-media-core');
const MovableArea = reactifyWebComponent('taro-movable-area-core');
const MovableView = reactifyWebComponent('taro-movable-view-core');
const PageContainer = reactifyWebComponent('taro-page-container-core');
const RootPortal = reactifyWebComponent('taro-root-portal-core');
const ScrollView = reactifyWebComponent('taro-scroll-view-core');
const ShareElement = reactifyWebComponent('taro-share-element-core');
const Swiper = reactifyWebComponent('taro-swiper-core');
const SwiperItem = reactifyWebComponent('taro-swiper-item-core');
const View = reactifyWebComponent('taro-view-core');
// 基础内容
const Icon = reactifyWebComponent('taro-icon-core');
const Progress = reactifyWebComponent('taro-progress-core');
const RichText = reactifyWebComponent('taro-rich-text-core');
const Text = reactifyWebComponent('taro-text-core');
// 表单组件
const Button = reactifyWebComponent('taro-button-core');
const Checkbox = reactifyWebComponent('taro-checkbox-core');
const CheckboxGroup = reactifyWebComponent('taro-checkbox-group-core');
const Editor = reactifyWebComponent('taro-editor-core');
const Form = reactifyWebComponent('taro-form-core');
const KeyboardAccessory = reactifyWebComponent('taro-keyboard-accessory-core');
const Label = reactifyWebComponent('taro-label-core');
const Picker = reactifyWebComponent('taro-picker-core');
const PickerView = reactifyWebComponent('taro-picker-view-core');
const PickerViewColumn = reactifyWebComponent('taro-picker-view-column-core');
const Radio = reactifyWebComponent('taro-radio-core');
const RadioGroup = reactifyWebComponent('taro-radio-group-core');
const Slider = reactifyWebComponent('taro-slider-core');
const Switch = reactifyWebComponent('taro-switch-core');
const Textarea = reactifyWebComponent('taro-textarea-core');
// 导航
const FunctionalPageNavigator = reactifyWebComponent('taro-functional-page-navigator-core');
const Navigator = reactifyWebComponent('taro-navigator-core');
const NavigationBar = reactifyWebComponent('taro-navigation-bar-core');
// 媒体组件
const Audio = reactifyWebComponent('taro-audio-core');
const Camera = reactifyWebComponent('taro-camera-core');
const Image = reactifyWebComponent('taro-image-core');
const LivePlayer = reactifyWebComponent('taro-live-player-core');
const LivePusher = reactifyWebComponent('taro-live-pusher-core');
const Video = reactifyWebComponent('taro-video-core');
const VoipRoom = reactifyWebComponent('taro-voip-room-core');
// 地图
const Map = reactifyWebComponent('taro-map-core');
// 画布
const Canvas = reactifyWebComponent('taro-canvas-core');
// 开放能力
const Ad = reactifyWebComponent('taro-ad-core');
const AdCustom = reactifyWebComponent('taro-ad-custom-core');
const OfficialAccount = reactifyWebComponent('taro-official-account-core');
const OpenData = reactifyWebComponent('taro-open-data-core');
const WebView = reactifyWebComponent('taro-web-view-core');
// 页面属性配置节点
const PageMeta = reactifyWebComponent('taro-page-meta-core');
// 其他
const Block = Fragment;
const CustomWrapper = reactifyWebComponent('taro-custom-wrapper-core');

export { Ad, AdCustom, Audio, Block, Button, Camera, Canvas, Checkbox, CheckboxGroup, CoverImage, CoverView, CustomWrapper, Editor, Form, FunctionalPageNavigator, Icon, Image, KeyboardAccessory, Label, LivePlayer, LivePusher, Map, MatchMedia, MovableArea, MovableView, NavigationBar, Navigator, OfficialAccount, OpenData, PageContainer, PageMeta, Picker, PickerView, PickerViewColumn, Progress, Radio, RadioGroup, RichText, RootPortal, ScrollView, ShareElement, Slider, Swiper, SwiperItem, Switch, Text, Textarea, Video, View, VoipRoom, WebView };
