// @tarojs/taro-h5/dist/api/media/voip.js
import { temporarilyNotSupport } from '../../utils/index.js';

// 实时语音
const updateVoIPChatMuteConfig = temporarilyNotSupport('updateVoIPChatMuteConfig');
const subscribeVoIPVideoMembers = temporarilyNotSupport('subscribeVoIPVideoMembers');
const setEnable1v1Chat = temporarilyNotSupport('setEnable1v1Chat');
const onVoIPVideoMembersChanged = temporarilyNotSupport('onVoIPVideoMembersChanged');
const onVoIPChatStateChanged = temporarilyNotSupport('onVoIPChatStateChanged');
const onVoIPChatSpeakersChanged = temporarilyNotSupport('onVoIPChatSpeakersChanged');
const onVoIPChatMembersChanged = temporarilyNotSupport('onVoIPChatMembersChanged');
const onVoIPChatInterrupted = temporarilyNotSupport('onVoIPChatInterrupted');
const offVoIPVideoMembersChanged = temporarilyNotSupport('offVoIPVideoMembersChanged');
const offVoIPChatStateChanged = temporarilyNotSupport('offVoIPChatStateChanged');
const offVoIPChatMembersChanged = temporarilyNotSupport('offVoIPChatMembersChanged');
const offVoIPChatInterrupted = temporarilyNotSupport('offVoIPChatInterrupted');
const joinVoIPChat = temporarilyNotSupport('joinVoIPChat');
const exitVoIPChat = temporarilyNotSupport('exitVoIPChat');

export { exitVoIPChat, joinVoIPChat, offVoIPChatInterrupted, offVoIPChatMembersChanged, offVoIPChatStateChanged, offVoIPVideoMembersChanged, onVoIPChatInterrupted, onVoIPChatMembersChanged, onVoIPChatSpeakersChanged, onVoIPChatStateChanged, onVoIPVideoMembersChanged, setEnable1v1Chat, subscribeVoIPVideoMembers, updateVoIPChatMuteConfig };
