// @tarojs/components/lib/vue2/components.js
import Vue from 'vue';
import { createCommonRender, createCommonMethod } from './vue-component-lib/utils.js';

/* eslint-disable */
const customElementTags = [
    'taro-ad-core',
    'taro-ad-custom-core',
    'taro-animation-video-core',
    'taro-animation-view-core',
    'taro-ar-camera-core',
    'taro-audio-core',
    'taro-aweme-data-core',
    'fragment',
    'taro-button-core',
    'taro-camera-core',
    'taro-canvas-core',
    'taro-channel-live-core',
    'taro-channel-video-core',
    'taro-checkbox-core',
    'taro-checkbox-group-core',
    'taro-comment-detail-core',
    'taro-comment-list-core',
    'taro-contact-button-core',
    'taro-cover-image-core',
    'taro-cover-view-core',
    'fragment',
    'taro-editor-core',
    'taro-follow-swan-core',
    'taro-form-core',
    'taro-functional-page-navigator-core',
    'taro-icon-core',
    'taro-image-core',
    'taro-inline-payment-panel-core',
    'taro-input-core',
    'taro-keyboard-accessory-core',
    'taro-label-core',
    'taro-lifestyle-core',
    'taro-like-core',
    'taro-live-player-core',
    'taro-live-pusher-core',
    'taro-login-core',
    'taro-lottie-core',
    'taro-map-core',
    'taro-match-media-core',
    'taro-movable-area-core',
    'taro-movable-view-core',
    'taro-native-slot-core',
    'taro-navigation-bar-core',
    'taro-navigator-core',
    'taro-official-account-core',
    'taro-open-data-core',
    'taro-page-container-core',
    'taro-page-meta-core',
    'taro-picker-core',
    'taro-picker-group',
    'taro-picker-view-column-core',
    'taro-picker-view-core',
    'taro-progress-core',
    'taro-pull-to-refresh',
    'taro-radio-core',
    'taro-radio-group-core',
    'taro-rich-text-core',
    'taro-root-portal-core',
    'taro-rtc-room-core',
    'taro-rtc-room-item-core',
    'taro-scroll-view-core',
    'taro-share-element-core',
    'taro-slider-core',
    'taro-slot-core',
    'taro-swiper-core',
    'taro-swiper-item-core',
    'taro-switch-core',
    'taro-tab-item-core',
    'taro-tabbar',
    'taro-tabs-core',
    'taro-text-core',
    'taro-textarea-core',
    'taro-video-control',
    'taro-video-core',
    'taro-video-danmu',
    'taro-view-core',
    'taro-voip-room-core',
    'taro-web-view-core',
];
Vue.config.ignoredElements = [...Vue.config.ignoredElements, ...customElementTags];
const Ad = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-ad-core', []),
});
const AdCustom = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-ad-custom-core', []),
});
const AnimationVideo = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-animation-video-core', []),
});
const AnimationView = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-animation-view-core', []),
});
const ArCamera = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-ar-camera-core', []),
});
const Audio = /*@__PURE__*/ Vue.extend({
    props: {
        src: {},
        controls: {},
        autoplay: {},
        loop: {},
        muted: {},
        nativeProps: {},
    },
    render: createCommonRender('taro-audio-core', ['error', 'play', 'pause', 'timeupdate', 'ended']),
});
const AwemeData = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-aweme-data-core', []),
});
const Block = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('fragment', []),
});
const Button = /*@__PURE__*/ Vue.extend({
    props: {
        disabled: {},
        hoverClass: {},
        type: {},
        hoverStartTime: {},
        hoverStayTime: {},
        size: {},
        plain: {},
        loading: {},
        formType: {},
    },
    render: createCommonRender('taro-button-core', ['tarobuttonsubmit', 'tarobuttonreset']),
});
const Camera = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-camera-core', []),
});
const Canvas = /*@__PURE__*/ Vue.extend({
    props: {
        canvasId: {},
        nativeProps: {},
    },
    render: createCommonRender('taro-canvas-core', ['longtap']),
});
const ChannelLive = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-channel-live-core', []),
});
const ChannelVideo = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-channel-video-core', []),
});
const Checkbox = /*@__PURE__*/ Vue.extend({
    props: {
        name: {},
        value: {},
        color: {},
        id: {},
        checked: {},
        disabled: {},
        nativeProps: {},
    },
    render: createCommonRender('taro-checkbox-core', ['checkboxchange']),
});
const CheckboxGroup = /*@__PURE__*/ Vue.extend({
    props: {
        name: {},
    },
    render: createCommonRender('taro-checkbox-group-core', ['change']),
});
const CommentDetail = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-comment-detail-core', []),
});
const CommentList = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-comment-list-core', []),
});
const ContactButton = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-contact-button-core', []),
});
const CoverImage = /*@__PURE__*/ Vue.extend({
    props: {
        src: {},
        nativeProps: {},
    },
    render: createCommonRender('taro-cover-image-core', ['load', 'error']),
});
const CoverView = /*@__PURE__*/ Vue.extend({
    props: {
        animation: {},
        hoverClass: {},
        hoverStartTime: {},
        hoverStayTime: {},
    },
    render: createCommonRender('taro-cover-view-core', ['longpress']),
});
const CustomWrapper = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('fragment', []),
});
const Editor = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-editor-core', []),
});
const FollowSwan = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-follow-swan-core', []),
});
const Form = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-form-core', ['submit']),
});
const FunctionalPageNavigator = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-functional-page-navigator-core', []),
});
const Icon = /*@__PURE__*/ Vue.extend({
    props: {
        type: {},
        size: {},
        color: {},
    },
    render: createCommonRender('taro-icon-core', []),
});
const Image = /*@__PURE__*/ Vue.extend({
    props: {
        src: {},
        mode: {},
        lazyLoad: {},
        nativeProps: {},
    },
    render: createCommonRender('taro-image-core', ['load', 'error']),
});
const InlinePaymentPanel = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-inline-payment-panel-core', []),
});
const Input = /*@__PURE__*/ Vue.extend({
    props: {
        value: {},
        type: {},
        password: {},
        placeholder: {},
        disabled: {},
        maxlength: {},
        autoFocus: {},
        confirmType: {},
        name: {},
        nativeProps: {},
    },
    model: {
        prop: 'value',
        event: 'update:modelValue'
    },
    methods: {
        focus: createCommonMethod('focus'),
    },
    render: createCommonRender('taro-input-core', ['input', 'paste', 'focus', 'blur', 'confirm', 'change', 'keydown']),
});
const KeyboardAccessory = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-keyboard-accessory-core', []),
});
const Label = /*@__PURE__*/ Vue.extend({
    props: {
        for: {},
    },
    render: createCommonRender('taro-label-core', []),
});
const Lifestyle = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-lifestyle-core', []),
});
const Like = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-like-core', []),
});
const LivePlayer = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-live-player-core', []),
});
const LivePusher = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-live-pusher-core', []),
});
const Login = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-login-core', []),
});
const Lottie = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-lottie-core', []),
});
const Map = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-map-core', []),
});
const MatchMedia = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-match-media-core', []),
});
const MovableArea = /*@__PURE__*/ Vue.extend({
    props: {
        scaleArea: {},
    },
    render: createCommonRender('taro-movable-area-core', []),
});
const MovableView = /*@__PURE__*/ Vue.extend({
    props: {
        x: {},
        y: {},
        direction: {},
        outOfBounds: {},
        inertia: {},
        friction: {},
        damping: {},
        disabled: {},
        scale: {},
        scaleMin: {},
        scaleMax: {},
        scaleValue: {},
        animation: {},
    },
    methods: {
        setParent: createCommonMethod('setParent'),
        endScale: createCommonMethod('endScale'),
        setScale: createCommonMethod('setScale'),
    },
    render: createCommonRender('taro-movable-view-core', ['change', 'scale', 'htouchmove', 'vtouchmove']),
});
const NativeSlot = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-native-slot-core', []),
});
const NavigationBar = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-navigation-bar-core', []),
});
const Navigator = /*@__PURE__*/ Vue.extend({
    props: {
        hoverClass: {},
        url: {},
        openType: {},
        isHover: {},
        delta: {},
    },
    render: createCommonRender('taro-navigator-core', ['cuccess', 'fail', 'complete']),
});
const OfficialAccount = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-official-account-core', []),
});
const OpenData = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-open-data-core', []),
});
const PageContainer = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-page-container-core', []),
});
const PageMeta = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-page-meta-core', []),
});
const Picker = /*@__PURE__*/ Vue.extend({
    props: {
        mode: {},
        disabled: {},
        range: {},
        rangeKey: {},
        value: {},
        start: {},
        end: {},
        fields: {},
        name: {},
    },
    model: {
        prop: 'value',
        event: 'update:modelValue'
    },
    render: createCommonRender('taro-picker-core', ['change', 'columnchange', 'cancel']),
});
const PickerGroup = /*@__PURE__*/ Vue.extend({
    props: {
        mode: {},
        range: {},
        rangeKey: {},
        height: {},
        columnId: {},
        updateHeight: {},
        onColumnChange: {},
        updateDay: {},
    },
    methods: {
        handleMoveStart: createCommonMethod('handleMoveStart'),
        handleMoving: createCommonMethod('handleMoving'),
        handleMoveEnd: createCommonMethod('handleMoveEnd'),
    },
    render: createCommonRender('taro-picker-group', []),
});
const PickerViewColumn = /*@__PURE__*/ Vue.extend({
    props: {
        col: {},
        initialPosition: {},
        paddingVertical: {},
    },
    render: createCommonRender('taro-picker-view-column-core', ['onselect', 'onselectstart', 'onselectend']),
});
const PickerView = /*@__PURE__*/ Vue.extend({
    props: {
        indicatorStyle: {},
        indicatorClass: {},
        value: {},
        maskStyle: {},
        maskClass: {},
    },
    render: createCommonRender('taro-picker-view-core', ['change', 'pickstart', 'pickend']),
});
const Progress = /*@__PURE__*/ Vue.extend({
    props: {
        percent: {},
        showInfo: {},
        borderRadius: {},
        fontSize: {},
        strokeWidth: {},
        activeColor: {},
        backgroundColor: {},
        active: {},
    },
    render: createCommonRender('taro-progress-core', []),
});
const PullToRefresh = /*@__PURE__*/ Vue.extend({
    props: {
        prefixCls: {},
        distanceToRefresh: {},
        damping: {},
        indicator: {},
    },
    render: createCommonRender('taro-pull-to-refresh', ['refresh']),
});
const Radio = /*@__PURE__*/ Vue.extend({
    props: {
        name: {},
        value: {},
        id: {},
        checked: {},
        disabled: {},
        nativeProps: {},
    },
    render: createCommonRender('taro-radio-core', ['radiochange']),
});
const RadioGroup = /*@__PURE__*/ Vue.extend({
    props: {
        name: {},
    },
    render: createCommonRender('taro-radio-group-core', ['change']),
});
const RichText = /*@__PURE__*/ Vue.extend({
    props: {
        nodes: {},
    },
    render: createCommonRender('taro-rich-text-core', []),
});
const RootPortal = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-root-portal-core', []),
});
const RtcRoom = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-rtc-room-core', []),
});
const RtcRoomItem = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-rtc-room-item-core', []),
});
const ScrollView = /*@__PURE__*/ Vue.extend({
    props: {
        scrollX: {},
        scrollY: {},
        upperThreshold: {},
        lowerThreshold: {},
        mpScrollTop: {},
        mpScrollLeft: {},
        mpScrollIntoView: {},
        animated: {},
    },
    methods: {
        mpScrollToMethod: createCommonMethod('mpScrollToMethod'),
        mpScrollIntoViewMethod: createCommonMethod('mpScrollIntoViewMethod'),
    },
    render: createCommonRender('taro-scroll-view-core', ['scroll', 'scrolltoupper', 'scrolltolower']),
});
const ShareElement = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-share-element-core', []),
});
const Slider = /*@__PURE__*/ Vue.extend({
    props: {
        min: {},
        max: {},
        step: {},
        disabled: {},
        value: {},
        activeColor: {},
        backgroundColor: {},
        blockSize: {},
        blockColor: {},
        showValue: {},
        name: {},
    },
    model: {
        prop: 'value',
        event: 'update:modelValue'
    },
    render: createCommonRender('taro-slider-core', ['change', 'changing']),
});
const Slot = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-slot-core', []),
});
const Swiper = /*@__PURE__*/ Vue.extend({
    props: {
        indicatorDots: {},
        indicatorColor: {},
        indicatorActiveColor: {},
        autoplay: {},
        current: {},
        interval: {},
        duration: {},
        circular: {},
        vertical: {},
        previousMargin: {},
        nextMargin: {},
        displayMultipleItems: {},
        full: {},
        zoom: {},
    },
    render: createCommonRender('taro-swiper-core', ['change', 'animationfinish']),
});
const SwiperItem = /*@__PURE__*/ Vue.extend({
    props: {
        itemId: {},
    },
    render: createCommonRender('taro-swiper-item-core', []),
});
const Switch = /*@__PURE__*/ Vue.extend({
    props: {
        type: {},
        checked: {},
        color: {},
        name: {},
        disabled: {},
        nativeProps: {},
    },
    model: {
        prop: 'checked',
        event: 'update:modelValue'
    },
    render: createCommonRender('taro-switch-core', ['change']),
});
const TabItem = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-tab-item-core', []),
});
const Tabbar = /*@__PURE__*/ Vue.extend({
    props: {
        conf: {},
    },
    render: createCommonRender('taro-tabbar', ['longpress']),
});
const Tabs = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-tabs-core', []),
});
const Text = /*@__PURE__*/ Vue.extend({
    props: {
        selectable: {},
    },
    render: createCommonRender('taro-text-core', []),
});
const Textarea = /*@__PURE__*/ Vue.extend({
    props: {
        value: {},
        placeholder: {},
        disabled: {},
        maxlength: {},
        autoFocus: {},
        autoHeight: {},
        name: {},
        nativeProps: {},
    },
    model: {
        prop: 'value',
        event: 'update:modelValue'
    },
    methods: {
        focus: createCommonMethod('focus'),
    },
    render: createCommonRender('taro-textarea-core', ['input', 'focus', 'blur', 'change', 'linechange']),
});
const VideoControl = /*@__PURE__*/ Vue.extend({
    props: {
        controls: {},
        currentTime: {},
        duration: {},
        isPlaying: {},
        pauseFunc: {},
        playFunc: {},
        seekFunc: {},
        showPlayBtn: {},
        showProgress: {},
    },
    methods: {
        setProgressBall: createCommonMethod('setProgressBall'),
        toggleVisibility: createCommonMethod('toggleVisibility'),
        getIsDraggingProgressBall: createCommonMethod('getIsDraggingProgressBall'),
        setCurrentTime: createCommonMethod('setCurrentTime'),
    },
    render: createCommonRender('taro-video-control', []),
});
const Video = /*@__PURE__*/ Vue.extend({
    props: {
        src: {},
        duration: {},
        controls: {},
        autoplay: {},
        loop: {},
        muted: {},
        initialTime: {},
        poster: {},
        objectFit: {},
        showProgress: {},
        showFullscreenBtn: {},
        showPlayBtn: {},
        showCenterPlayBtn: {},
        showMuteBtn: {},
        danmuList: {},
        danmuBtn: {},
        enableDanmu: {},
        enablePlayGesture: {},
        enableProgressGesture: {},
        vslideGesture: {},
        vslideGestureInFullscreen: {},
        nativeProps: {},
    },
    methods: {
        getHlsObject: createCommonMethod('getHlsObject'),
        play: createCommonMethod('play'),
        pause: createCommonMethod('pause'),
        stop: createCommonMethod('stop'),
        seek: createCommonMethod('seek'),
        requestFullScreen: createCommonMethod('requestFullScreen'),
        exitFullScreen: createCommonMethod('exitFullScreen'),
    },
    render: createCommonRender('taro-video-core', ['play', 'pause', 'ended', 'timeupdate', 'error', 'fullscreenchange', 'progress', 'loadedmetadata']),
});
const VideoDanmu = /*@__PURE__*/ Vue.extend({
    props: {
        enable: {},
    },
    methods: {
        sendDanmu: createCommonMethod('sendDanmu'),
        tick: createCommonMethod('tick'),
    },
    render: createCommonRender('taro-video-danmu', []),
});
const View = /*@__PURE__*/ Vue.extend({
    props: {
        animation: {},
        hoverClass: {},
        hoverStartTime: {},
        hoverStayTime: {},
    },
    render: createCommonRender('taro-view-core', ['longpress']),
});
const VoipRoom = /*@__PURE__*/ Vue.extend({
    render: createCommonRender('taro-voip-room-core', []),
});
const WebView = /*@__PURE__*/ Vue.extend({
    props: {
        src: {},
    },
    render: createCommonRender('taro-web-view-core', ['load', 'error']),
});

export { Ad, AdCustom, AnimationVideo, AnimationView, ArCamera, Audio, AwemeData, Block, Button, Camera, Canvas, ChannelLive, ChannelVideo, Checkbox, CheckboxGroup, CommentDetail, CommentList, ContactButton, CoverImage, CoverView, CustomWrapper, Editor, FollowSwan, Form, FunctionalPageNavigator, Icon, Image, InlinePaymentPanel, Input, KeyboardAccessory, Label, Lifestyle, Like, LivePlayer, LivePusher, Login, Lottie, Map, MatchMedia, MovableArea, MovableView, NativeSlot, NavigationBar, Navigator, OfficialAccount, OpenData, PageContainer, PageMeta, Picker, PickerGroup, PickerView, PickerViewColumn, Progress, PullToRefresh, Radio, RadioGroup, RichText, RootPortal, RtcRoom, RtcRoomItem, ScrollView, ShareElement, Slider, Slot, Swiper, SwiperItem, Switch, TabItem, Tabbar, Tabs, Text, Textarea, Video, VideoControl, VideoDanmu, View, VoipRoom, WebView };
