// @tarojs/components-react/dist/index.js
import './components/ad/loader.js';
export { Ad, AdCustom, AnimationVideo, AnimationView, ArCamera, Audio, AwemeData, Block, Button, Camera, Canvas, ChannelLive, ChannelVideo, Checkbox, CheckboxGroup, CommentDetail, CommentList, ContactButton, CoverImage, CoverView, CustomWrapper, Editor, FollowSwan, Form, FunctionalPageNavigator, Icon, InlinePaymentPanel, Input, KeyboardAccessory, Label, Lifestyle, Like, LivePlayer, LivePusher, Login, Lottie, Map, MatchMedia, MovableArea, MovableView, NativeSlot, NavigationBar, Navigator, OfficialAccount, OpenData, PageContainer, PageMeta, Picker, PickerGroup, PickerView, PickerViewColumn, Progress, Radio, RadioGroup, RichText, RootPortal, RtcRoom, RtcRoomItem, ShareElement, Slider, Slot, Switch, TabItem, Tabbar, Tabs, Textarea, Video, VideoControl, VideoDanmu, VoipRoom, WebView } from '@tarojs/components/lib/react';
import './components/ad-custom/loader.js';
import './components/animation-video/loader.js';
import './components/animation-view/loader.js';
import './components/ar-camera/loader.js';
import './components/audio/loader.js';
import './components/aweme-data/loader.js';
import './components/button/loader.js';
import './components/camera/loader.js';
import './components/canvas/loader.js';
import './components/channel-live/loader.js';
import './components/channel-video/loader.js';
import './components/checkbox/loader.js';
import './components/comment/loader.js';
import './components/contact-button/loader.js';
import './components/cover-image/loader.js';
import './components/cover-view/loader.js';
import './components/editor/loader.js';
import './components/follow-swan/loader.js';
import './components/form/loader.js';
import './components/functional-page-navigator/loader.js';
import './components/icon/loader.js';
export { default as Image } from './components/image/index.js';
import './components/inline-payment-panel/loader.js';
import './components/input/loader.js';
import './components/keyboard-accessory/loader.js';
import './components/label/loader.js';
import './components/lifestyle/loader.js';
import './components/like/loader.js';
import './components/live-player/loader.js';
import './components/live-pusher/loader.js';
import './components/login/loader.js';
import './components/lottie/loader.js';
import './components/map/loader.js';
import './components/match-media/loader.js';
import './components/movable-area/loader.js';
import './components/navigation-bar/loader.js';
import './components/navigator/loader.js';
import './components/official-account/loader.js';
import './components/open-data/loader.js';
import './components/page-container/loader.js';
import './components/page-meta/loader.js';
import './components/picker/loader.js';
import './components/picker-view/loader.js';
import './components/progress/loader.js';
export { default as PullDownRefresh } from './components/pull-down-refresh/index.js';
import './components/radio/loader.js';
import './components/rich-text/loader.js';
import './components/root-portal/loader.js';
import './components/rtc-room/loader.js';
export { default as ScrollView } from './components/scroll-view/index.js';
import './components/share-element/loader.js';
import './components/slider/loader.js';
import './components/slot/loader.js';
export { Swiper, SwiperItem } from './components/swiper/index.js';
import './components/switch/loader.js';
import './components/tabbar/loader.js';
import './components/tabs/loader.js';
export { default as Text } from './components/text/index.js';
import './components/textarea/loader.js';
import './components/video/loader.js';
export { default as View } from './components/view/index.js';
import './components/voip-room/loader.js';
import './components/web-view/loader.js';
