// @tarojs/taro-h5/dist/api/base/index.js
import { fromByteArray, toByteArray } from 'base64-js';
import { temporarilyNotSupport } from '../../utils/index.js';
export { getUserCryptoManager } from './crypto.js';
export { getLogManager, getRealtimeLogManager, setEnableDebug } from './debug.js';
export { getPerformance, reportPerformance } from './performance.js';
export { getAppAuthorizeSetting, getAppBaseInfo, getDeviceInfo, getSystemInfo, getSystemInfoAsync, getSystemInfoSync, getSystemSetting, getWindowInfo, openAppAuthorizeSetting, openSystemBluetoothSetting } from './system.js';
export { getUpdateManager, updateWeChatApp } from './update.js';
export { offAppHide, offAppShow, offAudioInterruptionBegin, offAudioInterruptionEnd, offError, offPageNotFound, offThemeChange, offUnhandledRejection, onAppHide, onAppShow, onAudioInterruptionBegin, onAudioInterruptionEnd, onError, onPageNotFound, onThemeChange, onUnhandledRejection } from './weapp/app-event.js';
export { getEnterOptionsSync, getLaunchOptionsSync } from './weapp/life-cycle.js';

const env = {
    FRAMEWORK: process.env.FRAMEWORK,
    TARO_ENV: process.env.TARO_ENV,
    TARO_VERSION: process.env.TARO_VERSION,
};
const canIUse = temporarilyNotSupport('canIUse');
function arrayBufferToBase64(arrayBuffer) {
    return fromByteArray(arrayBuffer);
}
function base64ToArrayBuffer(base64) {
    return toByteArray(base64);
}

export { arrayBufferToBase64, base64ToArrayBuffer, canIUse, env };
