// @tarojs/taro-h5/dist/api/location/index.js
import { temporarilyNotSupport, processOpenApi } from '../../utils/index.js';
export { getLocation } from './getLocation.js';
export { chooseLocation } from './chooseLocation.js';

// 位置
const stopLocationUpdate = temporarilyNotSupport('stopLocationUpdate');
const startLocationUpdateBackground = temporarilyNotSupport('startLocationUpdateBackground');
const startLocationUpdate = temporarilyNotSupport('startLocationUpdate');
const openLocation = processOpenApi({
    name: 'openLocation',
    defaultOptions: { scale: 18 }
});
const onLocationChangeError = temporarilyNotSupport('onLocationChangeError');
const onLocationChange = temporarilyNotSupport('onLocationChange');
const offLocationChangeError = temporarilyNotSupport('offLocationChangeError');
const offLocationChange = temporarilyNotSupport('offLocationChange');
const choosePoi = temporarilyNotSupport('choosePoi');
const getFuzzyLocation = temporarilyNotSupport('getFuzzyLocation');

export { choosePoi, getFuzzyLocation, offLocationChange, offLocationChangeError, onLocationChange, onLocationChangeError, openLocation, startLocationUpdate, startLocationUpdateBackground, stopLocationUpdate };
