// @tarojs/components/lib/vue3/component-lib/index.js
import createComponent from './createComponent.js';
import createFormsComponent from './createFormsComponent.js';
export { default as ScrollView } from './scroll-view.js';
export { default as Icon } from './icon.js';
export { default as Text } from './text.js';
export { default as Image } from './image.js';

// 视图容器
const CoverImage = createComponent('taro-cover-image');
const CoverView = createComponent('taro-cover-view');
const MatchMedia = createComponent('taro-match-media');
const MovableArea = createComponent('taro-movable-area');
const MovableView = createComponent('taro-movable-view');
const PageContainer = createComponent('taro-page-container');
const RootPortal = createComponent('taro-root-portal');
const ShareElement = createComponent('taro-share-element');
const Swiper = createComponent('taro-swiper');
const SwiperItem = createComponent('taro-swiper-item', ['swiper-slide']);
const View = createComponent('taro-view');
const Progress = createComponent('taro-progress', ['weui-progress']);
const RichText = createComponent('taro-rich-text');
// 表单组件
const Button = createComponent('taro-button');
const Checkbox = createComponent('taro-checkbox', ['weui-cells_checkbox']);
const CheckboxGroup = createComponent('taro-checkbox-group');
const Editor = createComponent('taro-editor');
const Form = createComponent('taro-form');
const Input = createFormsComponent('taro-input', 'input');
const KeyboardAccessory = createComponent('taro-keyboard-accessory');
const Label = createComponent('taro-label');
const Picker = createFormsComponent('taro-picker', 'change');
const PickerView = createComponent('taro-picker-view');
const PickerViewColumn = createComponent('taro-picker-view-column');
const Radio = createComponent('taro-radio', ['weui-cells_checkbox']);
const RadioGroup = createComponent('taro-radio-group', ['weui-cells_radiogroup']);
const Slider = createFormsComponent('taro-slider', 'change', 'value', ['weui-slider-box']);
const Switch = createFormsComponent('taro-switch', 'change', 'checked');
const Textarea = createFormsComponent('taro-textarea', 'input');
// 导航
const FunctionalPageNavigator = createComponent('taro-functional-page-navigator');
const Navigator = createComponent('taro-navigator');
// 媒体组件
const Audio = createComponent('taro-audio');
const Camera = createComponent('taro-camera');
const LivePlayer = createComponent('taro-live-player');
const Video = createComponent('taro-video', ['taro-video-container']);
const VoipRoom = createComponent('taro-voip-room');
// 地图
const Map = createComponent('taro-map');
// 画布
const Canvas = createComponent('taro-canvas');
// 开放能力
const WebView = createComponent('taro-web-view');
const Ad = createComponent('taro-ad');
const AdCustom = createComponent('taro-ad-custom');
const OfficialAccount = createComponent('taro-official-account');
const OpenData = createComponent('taro-open-data');
// 导航栏
const NavigationBar = createComponent('taro-navigation-bar');
// 页面属性配置节点
const PageMeta = createComponent('taro-page-meta');
// 其他
const Block = createComponent('taro-block');
const CustomWrapper = createComponent('taro-custom-wrapper');
const Slot = createComponent('taro-slot');

export { Ad, AdCustom, Audio, Block, Button, Camera, Canvas, Checkbox, CheckboxGroup, CoverImage, CoverView, CustomWrapper, Editor, Form, FunctionalPageNavigator, Input, KeyboardAccessory, Label, LivePlayer, Map, MatchMedia, MovableArea, MovableView, NavigationBar, Navigator, OfficialAccount, OpenData, PageContainer, PageMeta, Picker, PickerView, PickerViewColumn, Progress, Radio, RadioGroup, RichText, RootPortal, ShareElement, Slider, Slot, Swiper, SwiperItem, Switch, Textarea, Video, View, VoipRoom, WebView };
