// @tarojs/components/lib/vue3/components.js
import { defineComponent } from 'vue';
import { defineContainer } from './vue-component-lib/utils.js';

/* eslint-disable */
const Ad = /*@__PURE__*/ defineContainer('taro-ad-core', undefined);
const AdCustom = /*@__PURE__*/ defineContainer('taro-ad-custom-core', undefined);
const AnimationVideo = /*@__PURE__*/ defineContainer('taro-animation-video-core', undefined);
const AnimationView = /*@__PURE__*/ defineContainer('taro-animation-view-core', undefined);
const ArCamera = /*@__PURE__*/ defineContainer('taro-ar-camera-core', undefined);
const Audio = /*@__PURE__*/ defineContainer('taro-audio-core', undefined, [
    'src',
    'controls',
    'autoplay',
    'loop',
    'muted',
    'nativeProps',
    'error',
    'play',
    'pause',
    'timeupdate',
    'ended'
]);
const AwemeData = /*@__PURE__*/ defineContainer('taro-aweme-data-core', undefined);
const Block = defineComponent((__props, { slots }) => slots.default);
const Button = /*@__PURE__*/ defineContainer('taro-button-core', undefined, [
    'disabled',
    'hoverClass',
    'type',
    'hoverStartTime',
    'hoverStayTime',
    'size',
    'plain',
    'loading',
    'formType',
    'tarobuttonsubmit',
    'tarobuttonreset'
]);
const Camera = /*@__PURE__*/ defineContainer('taro-camera-core', undefined);
const Canvas = /*@__PURE__*/ defineContainer('taro-canvas-core', undefined, [
    'canvasId',
    'nativeProps',
    'longtap'
]);
const ChannelLive = /*@__PURE__*/ defineContainer('taro-channel-live-core', undefined);
const ChannelVideo = /*@__PURE__*/ defineContainer('taro-channel-video-core', undefined);
const Checkbox = /*@__PURE__*/ defineContainer('taro-checkbox-core', undefined, [
    'name',
    'value',
    'color',
    'id',
    'checked',
    'disabled',
    'nativeProps',
    'checkboxchange'
]);
const CheckboxGroup = /*@__PURE__*/ defineContainer('taro-checkbox-group-core', undefined, [
    'name',
    'change'
]);
const CommentDetail = /*@__PURE__*/ defineContainer('taro-comment-detail-core', undefined);
const CommentList = /*@__PURE__*/ defineContainer('taro-comment-list-core', undefined);
const ContactButton = /*@__PURE__*/ defineContainer('taro-contact-button-core', undefined);
const CoverImage = /*@__PURE__*/ defineContainer('taro-cover-image-core', undefined, [
    'src',
    'nativeProps',
    'load',
    'error'
]);
const CoverView = /*@__PURE__*/ defineContainer('taro-cover-view-core', undefined, [
    'animation',
    'hoverClass',
    'hoverStartTime',
    'hoverStayTime',
    'longpress'
]);
const CustomWrapper = defineComponent((__props, { slots }) => slots.default);
const Editor = /*@__PURE__*/ defineContainer('taro-editor-core', undefined);
const FollowSwan = /*@__PURE__*/ defineContainer('taro-follow-swan-core', undefined);
const Form = /*@__PURE__*/ defineContainer('taro-form-core', undefined, [
    'submit'
]);
const FunctionalPageNavigator = /*@__PURE__*/ defineContainer('taro-functional-page-navigator-core', undefined);
const Icon = /*@__PURE__*/ defineContainer('taro-icon-core', undefined, [
    'type',
    'size',
    'color'
]);
const Image = /*@__PURE__*/ defineContainer('taro-image-core', undefined, [
    'src',
    'mode',
    'lazyLoad',
    'nativeProps',
    'load',
    'error'
]);
const InlinePaymentPanel = /*@__PURE__*/ defineContainer('taro-inline-payment-panel-core', undefined);
const Input = /*@__PURE__*/ defineContainer('taro-input-core', undefined, [
    'value',
    'type',
    'password',
    'placeholder',
    'disabled',
    'maxlength',
    'autoFocus',
    'confirmType',
    'name',
    'nativeProps',
    'input',
    'paste',
    'focus',
    'blur',
    'confirm',
    'change',
    'keydown'
], 'value', 'input');
const KeyboardAccessory = /*@__PURE__*/ defineContainer('taro-keyboard-accessory-core', undefined);
const Label = /*@__PURE__*/ defineContainer('taro-label-core', undefined, [
    'for'
]);
const Lifestyle = /*@__PURE__*/ defineContainer('taro-lifestyle-core', undefined);
const Like = /*@__PURE__*/ defineContainer('taro-like-core', undefined);
const LivePlayer = /*@__PURE__*/ defineContainer('taro-live-player-core', undefined);
const LivePusher = /*@__PURE__*/ defineContainer('taro-live-pusher-core', undefined);
const Login = /*@__PURE__*/ defineContainer('taro-login-core', undefined);
const Lottie = /*@__PURE__*/ defineContainer('taro-lottie-core', undefined);
const Map = /*@__PURE__*/ defineContainer('taro-map-core', undefined);
const MatchMedia = /*@__PURE__*/ defineContainer('taro-match-media-core', undefined);
const MovableArea = /*@__PURE__*/ defineContainer('taro-movable-area-core', undefined, [
    'scaleArea'
]);
const MovableView = /*@__PURE__*/ defineContainer('taro-movable-view-core', undefined, [
    'x',
    'y',
    'direction',
    'outOfBounds',
    'inertia',
    'friction',
    'damping',
    'disabled',
    'scale',
    'scaleMin',
    'scaleMax',
    'scaleValue',
    'animation',
    'change',
    'scale',
    'htouchmove',
    'vtouchmove'
]);
const NativeSlot = /*@__PURE__*/ defineContainer('taro-native-slot-core', undefined);
const NavigationBar = /*@__PURE__*/ defineContainer('taro-navigation-bar-core', undefined);
const Navigator = /*@__PURE__*/ defineContainer('taro-navigator-core', undefined, [
    'hoverClass',
    'url',
    'openType',
    'isHover',
    'delta',
    'cuccess',
    'fail',
    'complete'
]);
const OfficialAccount = /*@__PURE__*/ defineContainer('taro-official-account-core', undefined);
const OpenData = /*@__PURE__*/ defineContainer('taro-open-data-core', undefined);
const PageContainer = /*@__PURE__*/ defineContainer('taro-page-container-core', undefined);
const PageMeta = /*@__PURE__*/ defineContainer('taro-page-meta-core', undefined);
const Picker = /*@__PURE__*/ defineContainer('taro-picker-core', undefined, [
    'mode',
    'disabled',
    'range',
    'rangeKey',
    'value',
    'start',
    'end',
    'fields',
    'name',
    'change',
    'columnchange',
    'cancel'
], 'value', 'change');
const PickerGroup = /*@__PURE__*/ defineContainer('taro-picker-group', undefined, [
    'mode',
    'range',
    'rangeKey',
    'height',
    'columnId',
    'updateHeight',
    'onColumnChange',
    'updateDay'
]);
const PickerViewColumn = /*@__PURE__*/ defineContainer('taro-picker-view-column-core', undefined, [
    'col',
    'initialPosition',
    'paddingVertical',
    'onselect',
    'onselectstart',
    'onselectend'
]);
const PickerView = /*@__PURE__*/ defineContainer('taro-picker-view-core', undefined, [
    'indicatorStyle',
    'indicatorClass',
    'value',
    'maskStyle',
    'maskClass',
    'change',
    'pickstart',
    'pickend'
]);
const Progress = /*@__PURE__*/ defineContainer('taro-progress-core', undefined, [
    'percent',
    'showInfo',
    'borderRadius',
    'fontSize',
    'strokeWidth',
    'activeColor',
    'backgroundColor',
    'active'
]);
const PullToRefresh = /*@__PURE__*/ defineContainer('taro-pull-to-refresh', undefined, [
    'prefixCls',
    'distanceToRefresh',
    'damping',
    'indicator',
    'refresh'
]);
const Radio = /*@__PURE__*/ defineContainer('taro-radio-core', undefined, [
    'name',
    'value',
    'id',
    'checked',
    'disabled',
    'nativeProps',
    'radiochange'
]);
const RadioGroup = /*@__PURE__*/ defineContainer('taro-radio-group-core', undefined, [
    'name',
    'change'
]);
const RichText = /*@__PURE__*/ defineContainer('taro-rich-text-core', undefined, [
    'nodes'
]);
const RootPortal = /*@__PURE__*/ defineContainer('taro-root-portal-core', undefined);
const RtcRoom = /*@__PURE__*/ defineContainer('taro-rtc-room-core', undefined);
const RtcRoomItem = /*@__PURE__*/ defineContainer('taro-rtc-room-item-core', undefined);
const ScrollView = /*@__PURE__*/ defineContainer('taro-scroll-view-core', undefined, [
    'scrollX',
    'scrollY',
    'upperThreshold',
    'lowerThreshold',
    'mpScrollTop',
    'mpScrollLeft',
    'mpScrollIntoView',
    'animated',
    'scroll',
    'scrolltoupper',
    'scrolltolower'
]);
const ShareElement = /*@__PURE__*/ defineContainer('taro-share-element-core', undefined);
const Slider = /*@__PURE__*/ defineContainer('taro-slider-core', undefined, [
    'min',
    'max',
    'step',
    'disabled',
    'value',
    'activeColor',
    'backgroundColor',
    'blockSize',
    'blockColor',
    'showValue',
    'name',
    'change',
    'changing'
], 'value', 'change');
const Slot = /*@__PURE__*/ defineContainer('taro-slot-core', undefined);
const Swiper = /*@__PURE__*/ defineContainer('taro-swiper-core', undefined, [
    'indicatorDots',
    'indicatorColor',
    'indicatorActiveColor',
    'autoplay',
    'current',
    'interval',
    'duration',
    'circular',
    'vertical',
    'previousMargin',
    'nextMargin',
    'displayMultipleItems',
    'full',
    'zoom',
    'change',
    'animationfinish'
]);
const SwiperItem = /*@__PURE__*/ defineContainer('taro-swiper-item-core', undefined, [
    'itemId'
]);
const Switch = /*@__PURE__*/ defineContainer('taro-switch-core', undefined, [
    'type',
    'checked',
    'color',
    'name',
    'disabled',
    'nativeProps',
    'change'
], 'checked', 'change');
const TabItem = /*@__PURE__*/ defineContainer('taro-tab-item-core', undefined);
const Tabbar = /*@__PURE__*/ defineContainer('taro-tabbar', undefined, [
    'conf',
    'longpress'
]);
const Tabs = /*@__PURE__*/ defineContainer('taro-tabs-core', undefined);
const Text = /*@__PURE__*/ defineContainer('taro-text-core', undefined, [
    'selectable'
]);
const Textarea = /*@__PURE__*/ defineContainer('taro-textarea-core', undefined, [
    'value',
    'placeholder',
    'disabled',
    'maxlength',
    'autoFocus',
    'autoHeight',
    'name',
    'nativeProps',
    'input',
    'focus',
    'blur',
    'change',
    'linechange'
], 'value', 'input');
const VideoControl = /*@__PURE__*/ defineContainer('taro-video-control', undefined, [
    'controls',
    'currentTime',
    'duration',
    'isPlaying',
    'pauseFunc',
    'playFunc',
    'seekFunc',
    'showPlayBtn',
    'showProgress'
]);
const Video = /*@__PURE__*/ defineContainer('taro-video-core', undefined, [
    'src',
    'duration',
    'controls',
    'autoplay',
    'loop',
    'muted',
    'initialTime',
    'poster',
    'objectFit',
    'showProgress',
    'showFullscreenBtn',
    'showPlayBtn',
    'showCenterPlayBtn',
    'showMuteBtn',
    'danmuList',
    'danmuBtn',
    'enableDanmu',
    'enablePlayGesture',
    'enableProgressGesture',
    'vslideGesture',
    'vslideGestureInFullscreen',
    'nativeProps',
    'play',
    'pause',
    'ended',
    'timeupdate',
    'error',
    'fullscreenchange',
    'progress',
    'loadedmetadata'
]);
const VideoDanmu = /*@__PURE__*/ defineContainer('taro-video-danmu', undefined, [
    'enable'
]);
const View = /*@__PURE__*/ defineContainer('taro-view-core', undefined, [
    'animation',
    'hoverClass',
    'hoverStartTime',
    'hoverStayTime',
    'longpress'
]);
const VoipRoom = /*@__PURE__*/ defineContainer('taro-voip-room-core', undefined);
const WebView = /*@__PURE__*/ defineContainer('taro-web-view-core', undefined, [
    'src',
    'load',
    'error'
]);

export { Ad, AdCustom, AnimationVideo, AnimationView, ArCamera, Audio, AwemeData, Block, Button, Camera, Canvas, ChannelLive, ChannelVideo, Checkbox, CheckboxGroup, CommentDetail, CommentList, ContactButton, CoverImage, CoverView, CustomWrapper, Editor, FollowSwan, Form, FunctionalPageNavigator, Icon, Image, InlinePaymentPanel, Input, KeyboardAccessory, Label, Lifestyle, Like, LivePlayer, LivePusher, Login, Lottie, Map, MatchMedia, MovableArea, MovableView, NativeSlot, NavigationBar, Navigator, OfficialAccount, OpenData, PageContainer, PageMeta, Picker, PickerGroup, PickerView, PickerViewColumn, Progress, PullToRefresh, Radio, RadioGroup, RichText, RootPortal, RtcRoom, RtcRoomItem, ScrollView, ShareElement, Slider, Slot, Swiper, SwiperItem, Switch, TabItem, Tabbar, Tabs, Text, Textarea, Video, VideoControl, VideoDanmu, View, VoipRoom, WebView };
