// @tarojs/webpack5-runner/dist/utils/H5AppInstance.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const helper_1 = require("@tarojs/helper");
const lodash_1 = require("lodash");
const path_1 = __importDefault(require("path"));
class H5AppInstance {
    constructor(entry = {}, options = {}) {
        this.options = (0, lodash_1.defaults)(options || {}, {
            sourceDir: '',
            entryFileName: 'app',
            frameworkExts: helper_1.SCRIPT_EXT
        });
        this.entry = entry;
    }
    get appEntry() {
        const { entryFileName, sourceDir } = this.options;
        const appEntry = this.entry[entryFileName];
        if (!appEntry)
            return path_1.default.join(sourceDir, entryFileName);
        if (Array.isArray(appEntry)) {
            return appEntry.filter(item => path_1.default.basename(item, path_1.default.extname(item)) === entryFileName)[0];
        }
        else if (Array.isArray(appEntry.import)) {
            return appEntry.import.filter(item => path_1.default.basename(item, path_1.default.extname(item)) === entryFileName)[0];
        }
        return appEntry;
    }
    get appConfig() {
        if (!this.__appConfig) {
            const appConfigPath = this.getConfigFilePath(this.appEntry);
            const appConfig = (0, helper_1.readConfig)(appConfigPath);
            if ((0, helper_1.isEmptyObject)(appConfig)) {
                throw new Error('缺少 app 全局配置，请检查！');
            }
            this.__appConfig = appConfig;
        }
        return this.__appConfig;
    }
    get pages() {
        if (!this.__pages) {
            const appPages = this.appConfig.pages;
            if (!appPages || !appPages.length) {
                throw new Error('全局配置缺少 pages 字段，请检查！');
            }
            const { frameworkExts, sourceDir } = this.options;
            this.__pages = new Set([
                ...appPages.map(item => ({
                    name: item,
                    path: (0, helper_1.resolveMainFilePath)(path_1.default.join(sourceDir, item), frameworkExts)
                }))
            ]);
            this.getSubPackages();
        }
        return this.__pages;
    }
    getSubPackages() {
        const subPackages = this.appConfig.subPackages || this.appConfig.subpackages;
        const { frameworkExts, sourceDir } = this.options;
        if (subPackages && subPackages.length) {
            subPackages.forEach(item => {
                if (item.pages && item.pages.length) {
                    const root = item.root;
                    item.pages.forEach(page => {
                        var _a;
                        let pageItem = `${root}/${page}`;
                        pageItem = pageItem.replace(/\/{2,}/g, '/');
                        let hasPageIn = false;
                        this.pages.forEach(({ name }) => {
                            if (name === pageItem) {
                                hasPageIn = true;
                            }
                        });
                        if (!hasPageIn) {
                            const pagePath = (0, helper_1.resolveMainFilePath)(path_1.default.join(sourceDir, pageItem), frameworkExts);
                            this.pages.add({
                                name: pageItem,
                                path: pagePath
                            });
                            (_a = this.appConfig.pages) === null || _a === void 0 ? void 0 : _a.push(pageItem);
                        }
                    });
                }
            });
        }
    }
    get pagesConfigList() {
        if (!this.__pagesConfigList) {
            const list = new Map();
            const pages = this.pages;
            pages.forEach(({ name, path }) => {
                const pageConfigPath = this.getConfigFilePath(path);
                list.set(name, pageConfigPath);
            });
            this.__pagesConfigList = list;
        }
        return this.__pagesConfigList;
    }
    getConfigFilePath(filePath = '') {
        // console.log('filePath: ', filePath)
        return (0, helper_1.resolveMainFilePath)(`${filePath.replace(path_1.default.extname(filePath), '')}.config`);
    }
    getTargetFilePath(filePath, targetExtname) {
        const extname = path_1.default.extname(filePath);
        if (extname)
            return filePath.replace(extname, targetExtname);
        return filePath + targetExtname;
    }
}
exports.default = H5AppInstance;
