// @tarojs/taro-h5/dist/api/media/background-audio/index.js
import { temporarilyNotSupport } from '../../../utils/index.js';
import { BackgroundAudioManager } from './BackgroundAudioManager.js';

// 背景音频
const stopBackgroundAudio = temporarilyNotSupport('stopBackgroundAudio');
const seekBackgroundAudio = temporarilyNotSupport('seekBackgroundAudio');
const playBackgroundAudio = temporarilyNotSupport('playBackgroundAudio');
const pauseBackgroundAudio = temporarilyNotSupport('pauseBackgroundAudio');
const onBackgroundAudioStop = temporarilyNotSupport('onBackgroundAudioStop');
const onBackgroundAudioPlay = temporarilyNotSupport('onBackgroundAudioPlay');
const onBackgroundAudioPause = temporarilyNotSupport('onBackgroundAudioPause');
const getBackgroundAudioPlayerState = temporarilyNotSupport('getBackgroundAudioPlayerState');
/**
 * 获取全局唯一的背景音频管理器
 */
const getBackgroundAudioManager = () => new BackgroundAudioManager();

export { getBackgroundAudioManager, getBackgroundAudioPlayerState, onBackgroundAudioPause, onBackgroundAudioPlay, onBackgroundAudioStop, pauseBackgroundAudio, playBackgroundAudio, seekBackgroundAudio, stopBackgroundAudio };
