// @tarojs/taro-h5/dist/api/swan/index.js
import { temporarilyNotSupport } from '../../utils/index.js';

const setPageInfo = temporarilyNotSupport('setPageInfo');
// 百度小程序 AI 相关
const ocrIdCard = temporarilyNotSupport('ocrIdCard');
const ocrBankCard = temporarilyNotSupport('ocrBankCard');
const ocrDrivingLicense = temporarilyNotSupport('ocrDrivingLicense');
const ocrVehicleLicense = temporarilyNotSupport('ocrVehicleLicense');
const textReview = temporarilyNotSupport('textReview');
const textToAudio = temporarilyNotSupport('textToAudio');
const imageAudit = temporarilyNotSupport('imageAudit');
const advancedGeneralIdentify = temporarilyNotSupport('advancedGeneralIdentify');
const objectDetectIdentify = temporarilyNotSupport('objectDetectIdentify');
const carClassify = temporarilyNotSupport('carClassify');
const dishClassify = temporarilyNotSupport('dishClassify');
const logoClassify = temporarilyNotSupport('logoClassify');
const animalClassify = temporarilyNotSupport('animalClassify');
const plantClassify = temporarilyNotSupport('plantClassify');
// 用户信息
const getSwanId = temporarilyNotSupport('getSwanId');
// 百度收银台支付
const requestPolymerPayment = temporarilyNotSupport('requestPolymerPayment');
// 打开小程序
const navigateToSmartGameProgram = temporarilyNotSupport('navigateToSmartGameProgram');
const navigateToSmartProgram = temporarilyNotSupport('navigateToSmartProgram');
const navigateBackSmartProgram = temporarilyNotSupport('navigateBackSmartProgram');
const preloadSubPackage = temporarilyNotSupport('preloadSubPackage');

export { advancedGeneralIdentify, animalClassify, carClassify, dishClassify, getSwanId, imageAudit, logoClassify, navigateBackSmartProgram, navigateToSmartGameProgram, navigateToSmartProgram, objectDetectIdentify, ocrBankCard, ocrDrivingLicense, ocrIdCard, ocrVehicleLicense, plantClassify, preloadSubPackage, requestPolymerPayment, setPageInfo, textReview, textToAudio };
