// @tarojs/taro-h5/dist/api/device/bluetooth-ble.js
import { temporarilyNotSupport } from '../../utils/index.js';

// 蓝牙-低功耗中心设备
const writeBLECharacteristicValue = temporarilyNotSupport('writeBLECharacteristicValue');
const setBLEMTU = temporarilyNotSupport('setBLEMTU');
const readBLECharacteristicValue = temporarilyNotSupport('readBLECharacteristicValue');
const onBLEMTUChange = temporarilyNotSupport('onBLEMTUChange');
const onBLEConnectionStateChange = temporarilyNotSupport('onBLEConnectionStateChange');
const onBLECharacteristicValueChange = temporarilyNotSupport('onBLECharacteristicValueChange');
const offBLEMTUChange = temporarilyNotSupport('offBLEMTUChange');
const offBLEConnectionStateChange = temporarilyNotSupport('offBLEConnectionStateChange');
const offBLECharacteristicValueChange = temporarilyNotSupport('offBLECharacteristicValueChange');
const notifyBLECharacteristicValueChange = temporarilyNotSupport('notifyBLECharacteristicValueChange');
const getBLEMTU = temporarilyNotSupport('getBLEMTU');
const getBLEDeviceServices = temporarilyNotSupport('getBLEDeviceServices');
const getBLEDeviceRSSI = temporarilyNotSupport('getBLEDeviceRSSI');
const getBLEDeviceCharacteristics = temporarilyNotSupport('getBLEDeviceCharacteristics');
const createBLEConnection = temporarilyNotSupport('createBLEConnection');
const closeBLEConnection = temporarilyNotSupport('closeBLEConnection');

export { closeBLEConnection, createBLEConnection, getBLEDeviceCharacteristics, getBLEDeviceRSSI, getBLEDeviceServices, getBLEMTU, notifyBLECharacteristicValueChange, offBLECharacteristicValueChange, offBLEConnectionStateChange, offBLEMTUChange, onBLECharacteristicValueChange, onBLEConnectionStateChange, onBLEMTUChange, readBLECharacteristicValue, setBLEMTU, writeBLECharacteristicValue };
