// @tarojs/taro-h5/dist/api/device/bluetooth.js
import { temporarilyNotSupport } from '../../utils/index.js';

// 蓝牙-通用
const stopBluetoothDevicesDiscovery = temporarilyNotSupport('stopBluetoothDevicesDiscovery');
const startBluetoothDevicesDiscovery = temporarilyNotSupport('startBluetoothDevicesDiscovery');
const openBluetoothAdapter = temporarilyNotSupport('openBluetoothAdapter');
const onBluetoothDeviceFound = temporarilyNotSupport('onBluetoothDeviceFound');
const onBluetoothAdapterStateChange = temporarilyNotSupport('onBluetoothAdapterStateChange');
const offBluetoothDeviceFound = temporarilyNotSupport('offBluetoothDeviceFound');
const offBluetoothAdapterStateChange = temporarilyNotSupport('offBluetoothAdapterStateChange');
const makeBluetoothPair = temporarilyNotSupport('makeBluetoothPair');
const isBluetoothDevicePaired = temporarilyNotSupport('isBluetoothDevicePaired');
const getConnectedBluetoothDevices = temporarilyNotSupport('getConnectedBluetoothDevices');
const getBluetoothDevices = temporarilyNotSupport('getBluetoothDevices');
const getBluetoothAdapterState = temporarilyNotSupport('getBluetoothAdapterState');
const closeBluetoothAdapter = temporarilyNotSupport('closeBluetoothAdapter');

export { closeBluetoothAdapter, getBluetoothAdapterState, getBluetoothDevices, getConnectedBluetoothDevices, isBluetoothDevicePaired, makeBluetoothPair, offBluetoothAdapterStateChange, offBluetoothDeviceFound, onBluetoothAdapterStateChange, onBluetoothDeviceFound, openBluetoothAdapter, startBluetoothDevicesDiscovery, stopBluetoothDevicesDiscovery };
