// @tarojs/taro-h5/dist/api/network/mdns.js
import { temporarilyNotSupport } from '../../utils/index.js';

// mDNS
const stopLocalServiceDiscovery = temporarilyNotSupport('stopLocalServiceDiscovery');
const startLocalServiceDiscovery = temporarilyNotSupport('startLocalServiceDiscovery');
const onLocalServiceResolveFail = temporarilyNotSupport('onLocalServiceResolveFail');
const onLocalServiceLost = temporarilyNotSupport('onLocalServiceLost');
const onLocalServiceFound = temporarilyNotSupport('onLocalServiceFound');
const onLocalServiceDiscoveryStop = temporarilyNotSupport('onLocalServiceDiscoveryStop');
const offLocalServiceResolveFail = temporarilyNotSupport('offLocalServiceResolveFail');
const offLocalServiceLost = temporarilyNotSupport('offLocalServiceLost');
const offLocalServiceFound = temporarilyNotSupport('offLocalServiceFound');
const offLocalServiceDiscoveryStop = temporarilyNotSupport('offLocalServiceDiscoveryStop');

export { offLocalServiceDiscoveryStop, offLocalServiceFound, offLocalServiceLost, offLocalServiceResolveFail, onLocalServiceDiscoveryStop, onLocalServiceFound, onLocalServiceLost, onLocalServiceResolveFail, startLocalServiceDiscovery, stopLocalServiceDiscovery };
